.github/workflows/darwin-tests.yaml

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2022 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Darwin Tests
+
+on:
+    push:
+    pull_request:
+    workflow_dispatch:
+
+concurrency:
+    group:
+        ${{ github.ref }}-${{ github.workflow }}-${{ (github.event_name ==
+        'pull_request' && github.event.number) || (github.event_name ==
+        'workflow_dispatch' && github.run_number) || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    test_suites_chip_tool_darwin:
+        name: Test Suites - Darwin
+        timeout-minutes: 120
+
+        strategy:
+            matrix:
+                build_variant: [no-ble-asan]
+        env:
+            BUILD_VARIANT: ${{matrix.build_variant}}
+
+        if: github.actor != 'restyled-io[bot]'
+        runs-on: macos-latest
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  submodules: true
```

Comment:
Will do! Thank you.

Reply:
I can't seem to find the script you are referring to, can you please point me in the right direction?

Reply:
See workflows from master, they have a step like:

```
 - name: Checkout submodules
              run: scripts/checkout_submodules.py --shallow --platform darwin
```

it may work as is or you may need to modify the script (likely it just works for darwin platform)

Reply:
Here for the current code:

https://github.com/project-chip/connectedhomeip/blob/master/.github/workflows/build.yaml#L300

Notice that checkout does not ask for submodules, but we have a separate steps for them.

Reply:
Ok. Updated.